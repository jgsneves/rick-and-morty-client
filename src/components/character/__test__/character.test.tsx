import { render } from '@testing-library/react';
import Character from '..';

describe('Character test suite', () => {
  it('should render only four eps', () => {
    const { asFragment } = render(
      <Character
        avatar="https//rickandmortyapi.com/api/character/avatar/1.jpeg"
        name="Rick Sanchez"
        origin="Earth (C-137)"
        location="Citadel of Ricks"
        epsodes={[
          'Pilot',
          'Lawnmower Dog',
          'Anatomy Park',
          'M. Night Shaym-Aliens!',
          'Meeseeks and Destroy',
          'Rick Potion #9',
          'Raising Gazorpazorp',
        ]}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render '+Mais' label", () => {
    const { getByText } = render(
      <Character
        avatar="https//rickandmortyapi.com/api/character/avatar/1.jpeg"
        name="Rick Sanchez"
        origin="Earth (C-137)"
        location="Citadel of Ricks"
        epsodes={[
          'Pilot',
          'Lawnmower Dog',
          'Anatomy Park',
          'M. Night Shaym-Aliens!',
          'Meeseeks and Destroy',
          'Rick Potion #9',
          'Raising Gazorpazorp',
        ]}
      />,
    );
    const label = getByText(
      /pilot - lawnmower dog - anatomy park - m\. night shaym-aliens!; \+mais/i,
    );
    expect(label).toBeInTheDocument();
  });
  it('should render only two eps', () => {
    const { asFragment } = render(
      <Character
        avatar="https//rickandmortyapi.com/api/character/avatar/1.jpeg"
        name="Rick Sanchez"
        origin="Earth (C-137)"
        location="Citadel of Ricks"
        epsodes={['Pilot', 'Lawnmower Dog']}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
