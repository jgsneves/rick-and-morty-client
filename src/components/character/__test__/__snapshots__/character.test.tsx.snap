// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character test suite should render only four eps 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hGjtOu"
  >
    <img
      class="sc-gsnTZi wtJol"
      src="https//rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <h2
      class="sc-dkzDqf bOavjn"
    >
      Rick Sanchez
    </h2>
    <p
      class="sc-hKMtZM fuiNwc"
    >
      Origem: 
    </p>
    <p
      class="sc-eCYdqJ kElERB"
    >
      Earth (C-137)
    </p>
    <p
      class="sc-hKMtZM fuiNwc"
    >
      Localização: 
    </p>
    <p
      class="sc-eCYdqJ kElERB"
    >
      Citadel of Ricks
    </p>
    <p
      class="sc-hKMtZM fuiNwc"
    >
      Episódios: 
    </p>
    <p
      class="sc-eCYdqJ kElERB"
    >
      Pilot - Lawnmower Dog - Anatomy Park - M. Night Shaym-Aliens!; +Mais
    </p>
  </div>
</DocumentFragment>
`;

exports[`Character test suite should render only two eps 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hGjtOu"
  >
    <img
      class="sc-gsnTZi wtJol"
      src="https//rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <h2
      class="sc-dkzDqf bOavjn"
    >
      Rick Sanchez
    </h2>
    <p
      class="sc-hKMtZM fuiNwc"
    >
      Origem: 
    </p>
    <p
      class="sc-eCYdqJ kElERB"
    >
      Earth (C-137)
    </p>
    <p
      class="sc-hKMtZM fuiNwc"
    >
      Localização: 
    </p>
    <p
      class="sc-eCYdqJ kElERB"
    >
      Citadel of Ricks
    </p>
    <p
      class="sc-hKMtZM fuiNwc"
    >
      Episódios: 
    </p>
    <p
      class="sc-eCYdqJ kElERB"
    >
      Pilot - Lawnmower Dog
    </p>
  </div>
</DocumentFragment>
`;
